File: src/array/cpp/1.cpp
#include <iostream>
#include <vector>
#include <cassert>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    result.push_back(i);
                    result.push_back(j);
                    return result;
                }
            }
        }
        return result;
    }
};


int main() {
    Solution s;
    vector<int> test1 = {2, 7, 11, 15};
    vector<int> result1 = s.twoSum(test1, 9);
    assert(result1[0] == 0 && result1[1] == 1 && "Test 1 Failed");

    vector<int> test2 = {3, 2, 4};
    vector<int> result2 = s.twoSum(test2, 6);
    assert(result2[0] == 1 && result2[1] == 2 && "Test 2 Failed");

    vector<int> test3 = {3, 3};
    vector<int> result3 = s.twoSum(test3, 6);
    assert(result3[0] == 0 && result3[1] == 1 && "Test 3 Failed");

    cout << "All tests passed!" << endl;
}

File: src/array/python/1.py
l = [2, 7, 11, 15]
l2 = [3, 2, 4]
l3 = [3, 3]

class Solution:
    def twoSum(self, nums, target):
        for i in range(0, len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return i, j
                

s = Solution()
print(s.twoSum(l, 9))
print(s.twoSum(l2, 6))
print(s.twoSum(l3, 6))
